pkg/kv/kvserver/liveness/liveness.go

```diff
@@ -641,13 +647,7 @@ func (nl *NodeLiveness) Start(ctx context.Context) {
 	nl.started.Set(true)
 	// We may have received some liveness records from Gossip prior to Start being
 	// called. We need to go through and notify all the callers of them now.
-	for _, entry := range nl.ScanNodeVitalityFromCache() {
-		if entry.IsLive(livenesspb.IsAliveNotification) {
-			for _, fn := range nl.onIsLive {
-				fn(entry.GetInternalLiveness())
-			}
-		}
-	}
+	nl.notifyIsAliveCallbacks()
```

Comment:
This doesn't actually notify anything, it needs to pass in the callbacks, or the method needs to collect them. This also reveals lacking test coverage, can we add a quick test?

Reply:
It's now tested, and I removed the footgun (need to pass a `[]IsLiveCallback`). This required some refactoring, basically picking up where @andrewbaptist left off at some point in the past (thanks for the initial leg work!)